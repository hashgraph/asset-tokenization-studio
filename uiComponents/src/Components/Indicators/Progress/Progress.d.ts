/// <reference types="react" />
import type { BaseMultiStyleConfiguration, PartsStyleInterpolation } from "@/Theme/Components/BaseMultiStyleConfiguration";
import type { progressAnatomy as ChakraProgressAnatomy } from "@chakra-ui/anatomy";
import type { ProgressProps as ChakraProgressProps } from "@chakra-ui/progress";
import type { SystemStyleObject as ChakraSystemStyleObject } from "@chakra-ui/system";
export declare const progressPartsList: Parts;
type Parts = typeof ChakraProgressAnatomy.keys;
export interface ProgressProps extends Omit<ChakraProgressProps, "colorScheme"> {
}
type ProgressThemeStyle = Record<Parts[number], ChakraSystemStyleObject>;
export interface ProgressThemeConfiguration extends BaseMultiStyleConfiguration<Parts> {
    baseStyle: (({ isIndeterminate, }: {
        isIndeterminate: boolean;
    }) => Partial<ProgressThemeStyle>) | PartsStyleInterpolation<Parts>;
}
export declare const Progress: (props: ProgressProps) => JSX.Element;
export {};
