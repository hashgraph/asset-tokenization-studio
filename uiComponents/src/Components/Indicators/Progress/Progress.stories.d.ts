declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("./Progress").ProgressProps>;
export default meta;
export declare const ExtraSmall: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Progress").ProgressProps>;
export declare const Small: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Progress").ProgressProps>;
export declare const Medium: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Progress").ProgressProps>;
export declare const Large: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Progress").ProgressProps>;
