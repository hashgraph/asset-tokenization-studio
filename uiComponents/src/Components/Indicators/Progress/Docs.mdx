import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as ProgressStories from "./Progress.stories";

<Meta of={ProgressStories} />

<Title />

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `Progress.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Progress: ConfigProgress,
  },
};
```

```tsx dark
const ConfigProgress: ProgressThemeConfiguration = {
  parts: progressPartsList,
  baseStyle: {},
  defaultProps: {},
};
```

### Component parts

You have these parts available: `"label" | "track" | "filledTrack"` (`progressPartsList`)

## Props

<ArgTypes of={ProgressStories} />

## Stories

<Stories title="" />
