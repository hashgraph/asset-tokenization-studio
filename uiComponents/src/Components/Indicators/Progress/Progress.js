var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { Progress as ChakraProgress } from "@chakra-ui/progress";
import React from "react";
export var progressPartsList = ["label", "track", "filledTrack"];
export var Progress = function (props) {
    return React.createElement(ChakraProgress, __assign({}, props));
};
