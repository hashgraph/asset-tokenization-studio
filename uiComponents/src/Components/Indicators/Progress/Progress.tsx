import type {
  BaseMultiStyleConfiguration,
  PartsStyleInterpolation,
} from "@/Theme/Components/BaseMultiStyleConfiguration";
import type { progressAnatomy as ChakraProgressAnatomy } from "@chakra-ui/anatomy";
import type { ProgressProps as ChakraProgressProps } from "@chakra-ui/progress";
import { Progress as ChakraProgress } from "@chakra-ui/progress";
import type { SystemStyleObject as ChakraSystemStyleObject } from "@chakra-ui/system";
import React from "react";

export const progressPartsList: Parts = ["label", "track", "filledTrack"];

type Parts = typeof ChakraProgressAnatomy.keys;

export interface ProgressProps
  extends Omit<ChakraProgressProps, "colorScheme"> {}

type ProgressThemeStyle = Record<Parts[number], ChakraSystemStyleObject>;

export interface ProgressThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (({
        isIndeterminate,
      }: {
        isIndeterminate: boolean;
      }) => Partial<ProgressThemeStyle>)
    | PartsStyleInterpolation<Parts>;
}

export const Progress = (props: ProgressProps) => {
  return <ChakraProgress {...props}></ChakraProgress>;
};
