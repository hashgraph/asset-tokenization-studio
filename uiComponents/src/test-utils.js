import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
