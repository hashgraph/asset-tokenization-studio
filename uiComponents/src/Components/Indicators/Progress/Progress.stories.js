var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Progress } from "./Progress";
var meta = {
    title: "Design System/Indicators/Progress",
    component: Progress,
    argTypes: {
        value: { control: "number" },
        max: { control: "number" },
        min: { control: "number" },
        size: { control: "select", options: ["xs", "sm", "md", "lg"] },
        hasStripe: { control: "boolean" },
        isAnimated: { control: "boolean" },
        isIndeterminate: { control: "boolean" }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A10606"
        },
        docs: {}
    },
    args: {
        max: 100,
        min: 0,
        value: 60,
        size: "sm",
        hasStripe: false,
        isAnimated: false,
        isIndeterminate: false
    }
};
export default meta;
var Template = function (args) {
    return React.createElement(Progress, __assign({}, args));
};
export var ExtraSmall = Template.bind({});
ExtraSmall.args = {
    value: 30,
    size: "xs"
};
export var Small = Template.bind({});
Small.args = {
    value: 30,
    size: "sm"
};
export var Medium = Template.bind({});
Medium.args = {
    value: 30,
    size: "md"
};
export var Large = Template.bind({});
Large.args = {
    value: 30,
    size: "lg"
};
