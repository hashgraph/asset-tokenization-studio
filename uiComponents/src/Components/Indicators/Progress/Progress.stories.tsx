import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { Progress } from "./Progress";

const meta = {
  title: "Design System/Indicators/Progress",
  component: Progress,
  argTypes: {
    value: { control: "number" },
    max: { control: "number" },
    min: { control: "number" },
    size: { control: "select", options: ["xs", "sm", "md", "lg"] },
    hasStripe: { control: "boolean" },
    isAnimated: { control: "boolean" },
    isIndeterminate: { control: "boolean" },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A10606",
    },
    docs: {},
  },
  args: {
    max: 100,
    min: 0,
    value: 60,
    size: "sm",
    hasStripe: false,
    isAnimated: false,
    isIndeterminate: false,
  },
} as Meta<typeof Progress>;
export default meta;

const Template: StoryFn<typeof Progress> = (args) => {
  return <Progress {...args}></Progress>;
};

export const ExtraSmall = Template.bind({});
ExtraSmall.args = {
  value: 30,
  size: "xs",
};

export const Small = Template.bind({});
Small.args = {
  value: 30,
  size: "sm",
};

export const Medium = Template.bind({});
Medium.args = {
  value: 30,
  size: "md",
};

export const Large = Template.bind({});
Large.args = {
  value: 30,
  size: "lg",
};
