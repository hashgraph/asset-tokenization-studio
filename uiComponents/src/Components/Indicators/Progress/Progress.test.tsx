import { render } from "@/test-utils";
import React from "react";
import type { ProgressProps } from "./Progress";
import { Progress } from "./Progress";

describe(`<Progress />`, () => {
  const value = 60;

  const min = 0;

  const max = 100;

  const hasStripe = false;

  const isAnimated = false;

  const isIndeterminate = false;

  const defaultProps = {
    value,
    min,
    max,
    hasStripe,
    isAnimated,
    isIndeterminate,
  };

  const factoryComponent = (props: ProgressProps) =>
    render(<Progress {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent({ ...defaultProps });

    const progressStyle = getComputedStyle(component.getByRole("progressbar"));
    expect(progressStyle.width).toBe(
      `${(defaultProps.value / (defaultProps.max - defaultProps.min)) * 100}%`
    );
    expect(component.asFragment()).toMatchSnapshot("Default progress");
  });

  test("renders correctly with stripes", () => {
    const component = factoryComponent({ ...defaultProps, hasStripe: true });

    const progressStyle = getComputedStyle(component.getByRole("progressbar"));
    expect(progressStyle.width).toBe(
      `${(defaultProps.value / (defaultProps.max - defaultProps.min)) * 100}%`
    );
    expect(component.asFragment()).toMatchSnapshot("Striped progress");
  });
  test("renders correctly with stripes and animation", () => {
    const component = factoryComponent({
      ...defaultProps,
      hasStripe: true,
      isAnimated: true,
    });

    const progressStyle = getComputedStyle(component.getByRole("progressbar"));
    expect(progressStyle.width).toBe(
      `${(defaultProps.value / (defaultProps.max - defaultProps.min)) * 100}%`
    );
    expect(component.asFragment()).toMatchSnapshot(
      "Striped progress with animation"
    );
  });
});
