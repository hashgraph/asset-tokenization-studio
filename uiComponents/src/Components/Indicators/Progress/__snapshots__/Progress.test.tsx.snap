// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> renders correctly with stripes and animation: Striped progress with animation 1`] = `
<DocumentFragment>
  <div
    class="css-9mbr6u"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="60"
      class="css-b2wzt2"
      role="progressbar"
      style="width: 60%;"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Progress /> renders correctly with stripes: Striped progress 1`] = `
<DocumentFragment>
  <div
    class="css-9mbr6u"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="60"
      class="css-1ulqoxj"
      role="progressbar"
      style="width: 60%;"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Progress /> renders correctly: Default progress 1`] = `
<DocumentFragment>
  <div
    class="css-9mbr6u"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="60"
      class="css-1nsw7s9"
      role="progressbar"
      style="width: 60%;"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
